### PR TITLE
Skip trailing commas with FlowShorthandWithOneArg

### LINKED ISSUE #1363: Invalid code emitted with flow, default arguments and trailing commas
```js
class X {
  constructor(
    ideConnectionFactory: child_process$ChildProcess => FlowIDEConnection =
        defaultIDEConnectionFactory,
  ) {
  }
}
```

```js
class X {
  constructor(
    ideConnectionFactory:
      child_process$ChildProcess,
     => FlowIDEConnection = defaultIDEConnectionFactory,
  ) {}
}
```

```js
SyntaxError: Unexpected token (5:5)
  3 |     ideConnectionFactory:
  4 |       child_process$ChildProcess,
> 5 |      => FlowIDEConnection = defaultIDEConnectionFactory,
    |      ^
  6 |   ) {}
  7 | }
```

https://prettier.github.io/prettier/#%7B%22content%22%3A%22class%20X%20%7B%5Cn%20%20constructor(%5Cn%20%20%20%20ideConnectionFactory%3A%20child_process%24ChildProcess%20%3D%3E%20FlowIDEConnection%20%3D%5Cn%20%20%20%20%20%20%20%20defaultIDEConnectionFactory%2C%5Cn%20%20)%20%7B%5Cn%20%20%7D%5Cn%7D%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22all%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Afalse%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

### PR DESCRIPTION
**Pull request by existentialism**
Fixes #1363.

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -2372,7 +2372,11 @@ function printFunctionParams(path, print, options, expandArg) {
     isFlowShorthandWithOneArg ? "" : "(",
     indent(concat([softline, join(concat([",", line]), printed)])),
     ifBreak(
-      canHaveTrailingComma && shouldPrintComma(options, "all") ? "," : ""
+      !isFlowShorthandWithOneArg &&
```
Comment (vjeux) [inline src/printer.js]: Could you put this condition inside of `canHaveTrailingComma` instead?

**Comment by vjeux** [comment]
Awesome, thanks for fixing this! I'll merge it after you apply my small suggestion!

**Comment by existentialism** [comment]
👍

**Comment by vjeux** [comment]
Hmm I'm not sure if you updated it or I wasn't clear. I meant doing

```js
canHaveTrailingComma =
  !isFlowShorthandWithOneArg &&
  // existing conditions
```

this way you just have to check for `canHaveTrailingComma` inside of the place where you actually print the trailing comma.

**Comment by existentialism** [comment]
lol sorry, totally misread, updated!

**Pull request merged**